modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaOjectApioTest.java

```diff
@@ -0,0 +1,641 @@
+		).header(
+			"Accept", "application/hal+json"
+		).when(
+		).get(
+			documentsHref
+		).then(
+		).statusCode(
+			200
+		).body(
+			"_embedded.'Liferay:Document'.find {it.title == 'document.pdf'}." +
+				"contentUrl",
+			IsNull.notNullValue()
+		).body(
+			"_embedded.'Liferay:Document'.find {it.title == 'document.pdf'}." +
+				"dateCreated",
+			IsNull.notNullValue()
+		).body(
+			"_embedded.'Liferay:Document'.find {it.title == 'document.pdf'}." +
+				"dateModified",
+			IsNull.notNullValue()
+		).body(
+			"_embedded.'Liferay:Document'.find {it.title == 'document.pdf'}." +
+				"encodingFormat",
+			Matchers.equalTo("application/pdf")
+		).body(
+			"_embedded.'Liferay:Document'.find {it.title == 'document.pdf'}." +
+				"fileExtension",
+			Matchers.equalTo("pdf")
+		).body(
+			"_embedded.'Liferay:Document'.find {it.title == 'document.pdf'}." +
+				"keywords",
+			IsNull.notNullValue()
+		).body(
+			"_embedded.'Liferay:Document'.find {it.title == 'document.pdf'}." +
+				"sizeInBytes",
+			Matchers.greaterThan(0)
+		).body(
+			"_embedded.'Liferay:Document'.find {it.title == 'document.pdf'}." +
+				"title",
+			Matchers.equalTo("document.pdf")
+		).body(
+			"_embedded.'Liferay:Document'.find {it.title == 'document.pdf'}." +
+				"_links.category.href",
+			IsNull.notNullValue()
+		).body(
+			"_embedded.'Liferay:Document'.find {it.title == 'document.pdf'}." +
+				"_links.creator",
+			IsNull.notNullValue()
+		).body(
+			"_embedded.'Liferay:Document'.find {it.title == 'document.pdf'}." +
+				"_links.folder",
+			IsNull.notNullValue()
+		).body(
+			"_embedded.'Liferay:Document'.find {it.title == 'document.pdf'}." +
+				"_links.self",
+			IsNull.notNullValue()
+		);
+	}
+
+	@Test
+	public void testUpdateDocumentIsNotSupported() {
+		String documentsHref = ApioClientBuilder.given(
+		).basicAuth(
+			"test@liferay.com", "test"
+		).header(
+			"Accept", "application/hal+json"
+		).when(
+		).get(
+			_rootEndpointURL.toExternalForm()
+		).follow(
+			"_links.content-space.href"
+		).follow(
+			"_embedded.ContentSpace.find {it.name == '" +
+				MediaObjectTestActivator.CONTENT_SPACE_NAME +
+					"'}._links.documentsRepository.href"
+		).then(
+		).extract(
+		).path(
+			"_links.documents.href"
+		);
+
+		_documentHref = ApioClientBuilder.given(
+		).basicAuth(
+			"test@liferay.com", "test"
+		).header(
+			"Accept", "application/hal+json"
+		).header(
+			"Content-Type", "multipart/form-data"
+		).multipart(
+			"binaryFile", _readFile("document.pdf")
+		).when(
+		).post(
+			documentsHref
+		).then(
+		).statusCode(
+			200
+		).extract(
+		).path(
+			"_links.self.href"
+		);
+
+		ApioClientBuilder.given(
+		).basicAuth(
+			"test@liferay.com", "test"
+		).header(
+			"Accept", "application/hal+json"
+		).header(
+			"Content-Type", "multipart/form-data"
+		).multipart(
+			"binaryFile", _readFile("document-updated.pdf")
+		).when(
+		).put(
+			_documentHref
+		).then(
+		).statusCode(
+			404
```

Comment:
Don't test 404 error in functional test, test it as integration tests

Reply:
👍